projects/packages/connection/src/class-manager.php

```diff
@@ -972,6 +972,20 @@ public function register( $api_endpoint = 'register' ) {
 
 		$this->get_tokens()->update_blog_token( (string) $registration_details->jetpack_secret );
 
+		if ( empty( $registration_details->allow_inplace_authorization ) ) {
+			// Forces register_site REST endpoint to return the Calypso authorization URL.
+			add_filter( 'jetpack_use_iframe_authorization_flow', '__return_false', 20 );
+		}
+
+		add_filter(
+			'jetpack_register_site_rest_response',
+			function ( $response ) use ( $registration_details ) {
+				$response['allowInplaceAuthorization'] = $registration_details->allow_inplace_authorization;
```

Comment:
Should we handle the case when `allow_inplace_authorization` isn't set here?

Reply:
well, we control the server response and we are relying on other attributes without checking, such as `jetpack_id`, `jetpack_public` and `jetpack_secret`.

But thinking that this property is likely to become obsolete in the near future, adding a check would allow us to remove it from our responses without breaking backward compatibility... so yes, it might be a good idea!

Reply:
https://github.com/Automattic/jetpack/pull/19833/commits/ec69db748c9120c93feff134ab8c503949aecbe7